src/cli/parse-flags.ts

```diff
@@ -300,6 +305,21 @@ const setCLIArg = (flags: ConfigFlags, rawArg: string, normalizedArg: string, va
     }
   }
 
+  // We're setting a value which could be either a boolean _or_ a string
+  else if (readOnlyArrayHasStringMember(BOOLEAN_STRING_CLI_FLAGS, normalizedArg)) {
+    const derivedValue =
+      typeof value === 'string'
+        ? value
+          ? value // use the supplied value if it's a non-empty string
+          : false // otherwise, default to false for the empty string
+        : true; // no value was supplied, default to true
```

Comment:
nested ternaries make my head spin haha

basically we want to coerce `""` to `false`, keep any truthy string, and use `true` otherwise, since we handle the `--no-` case above?

Reply:
Yah, exactly. I couldn't come up with a "nicer" way to express this without doing something like:
```
let derivedValue = true;
if (typeof value === 'string') {
  if (value) {
    derivedValue = value;
  } else {
    derivedValue = false;
  }
}
```
which felt a little less than clear. IDK though, maybe that's just me (I also don't _love_ the ternary myself). LMK if you'd prefer the `if` statement style above.